filters/apiusagemonitoring/pathinfo.go

```diff
@@ -6,13 +6,22 @@ import (
 	"regexp"
 )
 
+type Regex struct{ *regexp.Regexp }
+
+func (r *Regex) MarshalJSON() ([]byte, error) {
+	if r == nil {
+		return json.Marshal(nil)
+	}
+	return json.Marshal(r.String())
+}
+
 // pathInfo contains the tracking information for a specific path.
 // The exported members are marshaled as JSON.
 type pathInfo struct {
 	ApplicationId  string
 	ApiId          string
 	PathTemplate   string
-	Matcher        *regexp.Regexp
+	Matcher        *Regex
 	ClientTracking *clientTrackingInfo
```

Comment:
wouldn't we need to use the same approach on the clientTrackingInfo type, which also has a Regexp field?